plugin/pkg/shoot/validator/admission.go

```diff
@@ -354,6 +354,26 @@ func (c *validationContext) validateProjectMembership(a admission.Attributes) er
 	return nil
 }
 
+// validateSeedSelectionForHAShoot validates the seed assigned for the shoot based on the following criteria:
+// 1. A HA shoot with failure tolerance of 'zone' can only be scheduled on multi-zonal seeds
+// 2. A HA shoot with failure tolerance of 'node' can technically be scheduled onto a multi-zonal as well as single-zonal seed. However, we restrict its scheduling only to multi-zonal seeds as of now.
```

Comment:
> However, we restrict its scheduling only to multi-zonal seeds as of now.

Extend the comment with a reasoning?

Reply:
This will change now. Need to confirm this with Vedran today. Will hold on to this for now.

Reply:
The check has been changed. Please check again.